content/Gems You Want To Have In Your Rails API - Juan Diego Rosales.md

```diff
@@ -0,0 +1,44 @@
+# Gems You Want To Have In Your Rails API
+
+*Juan Diego Rosales is a software developer on the Rootstrap team and writes about his experience working on different projects. You can also follow Juan on [LinkedIn](www.linkedin.com/in/juan-diego-rosales-02a05a144).*
+
+Are you new to Rails? Or maybe just looking for any cool gem to use in your new API? Here is a list of gems we have on our Rails API Base that we consider a must-have in every Rails API project.
+
+## Authentication
+If you are building a Rails API, then you will probably need token based authentication and that is when [Devise Token Auth](https://github.com/lynndylanhurley/devise_token_auth) comes in. This gem is built on top of Devise, one of the most popular authentication gems for Rails which you may have already worked with. If that is the case, then you'll get familiar with Devise Token Auth very easily.
+
+With very little setup, this gem provides registration, sign in, sign out and reset password flows out of the box, as well as very useful helper functions. Also, it handles email confirmation if enabled and supports the use of multiple user models, giving you the possibility to manage the authentication of different types of users at the same time using groups.
+
+In comparison to other alternatives available, one of the best perks that comes with Devise Token Auth, which is inherited from Devise, is the very little effort you have to invest to get things up and running. On top of that, you can also customize the different flows mentioned if you want to offer a different experience, and while this may be a little more challenging, it's not rocket science!
+
+## Resource administration
+Most APIs typically need some level of resource administration and for that I would definitely recommend [Active Admin](https://github.com/activeadmin/activeadmin). Despite the fact it can have a steep learning curve, specially at the beginning, due to it's DSL (domain-specific language), once you get used to it you can have admin pages with CRUD actions for any resource in no time.
```

Comment:
Suggestion: for someone who isn't familiar with the DSL acronym, it might be of help pointing to a summary of the concept rather than just unfurl it. It doesn't have to be the MF article, it could also be Wikipedia or any other resource that provides the needed clarification.

```suggestion
Most APIs typically need some level of resource administration and for that I would definitely recommend [Active Admin](https://github.com/activeadmin/activeadmin). Despite the fact it can have a steep learning curve, specially at the beginning, due to it's [DSL](https://martinfowler.com/dsl.html), once you get used to it you can have admin pages with CRUD actions for any resource in no time.
```

Reply:
Good suggestion! 👍